src/platform/plugins/shared/dashboard/public/dashboard_api/performance/query_performance_tracking.ts

```diff
@@ -123,8 +131,52 @@ function reportPerformanceMetrics({
 }) {
   const duration =
     loadType === 'dashboardSubsequentLoad' ? timeToData : Math.max(timeToData, totalLoadTime);
+  const groupedPerformanceMarkers = getPerformanceTrackersGroupedById(
+    PERFORMANCE_TRACKER_TYPES.PANEL
+  );
 
-  const e = {
+  // `groupedPerformanceMarkers` is a map of performance markers grouped by id.
+  // Each group contains the performance markers for a single Lens embeddable.
+  // We need to extract the start and end times of the preRender, renderStart
+  // and renderComplete markers and calculate the duration of each phase.
+  const measurements = Object.values(groupedPerformanceMarkers).map((group) => {
+    // Get the marker name from the first group.
+    const markerName =
+      Array.isArray(group) && group.length > 0
+        ? group[0].name.split(':').slice(0, -1).join(':')
+        : undefined;
+
+    const preRenderStart = group.find((marker) =>
+      marker.name.endsWith(`:${PERFORMANCE_TRACKER_MARKS.PRE_RENDER}`)
+    )?.startTime;
+    const renderStart = group.find((marker) =>
+      marker.name.endsWith(`:${PERFORMANCE_TRACKER_MARKS.RENDER_START}`)
+    )?.startTime;
+    const renderComplete = group.find((marker) =>
+      marker.name.endsWith(`:${PERFORMANCE_TRACKER_MARKS.RENDER_COMPLETE}`)
+    )?.startTime;
+
+    if (markerName && preRenderStart && renderStart) {
+      performance.measure(`${markerName}:${PERFORMANCE_TRACKER_MEASURES.PRE_RENDER_DURATION}`, {
+        start: preRenderStart,
+        end: renderStart,
+      });
+    }
+
+    if (markerName && renderComplete && renderStart) {
+      performance.measure(`${markerName}:${PERFORMANCE_TRACKER_MEASURES.RENDER_DURATION}`, {
+        start: renderStart,
+        end: renderComplete,
+      });
+    }
+
+    return {
```

Comment:
How about using `PERFORMANCE_TRACKER_MEASURES` constants instead of hard coding `preRenderDuration` and `renderDuration`.
```
return {
      [PERFORMANCE_TRACKER_MEASURES.PRE_RENDER_DURATION]: preRenderStart && renderStart ? renderStart - preRenderStart : 0,
      [PERFORMANCE_TRACKER_MEASURES.RENDER_DURATION]: renderComplete && renderStart ? renderComplete - renderStart : 0,
    };
```